### PR TITLE
Added some programs for CPP and C

### PR DESCRIPTION
**Pull request by AzeemIdrisi**
Improved some programs and added new programs. Please mark hacktoberfest-accepted during merge.

### CODE REVIEW

**Comment by sobhanbera** [comment]
Hi @AzeemIdrisi!

It very nice of you for the contibutions. But there are multiple files in this PR. Can you also attach the program's output so that it would be easy for me.

Thanks!

**Review by AzeemIdrisi — COMMENTED** [review]
Update: Attached output at the end of all programs. Please also add hacktoberfest-accepted tag while merging. Thank You.

**Comment by AzeemIdrisi** [comment]
@sobhanbera Check updates.

File: algorithms/cpp/binary_search.cpp [hunk 1]
```diff
@@ -1,13 +1,13 @@
 #include <bits/stdc++.h>
```
Comment (sobhanbera) [inline algorithms/cpp/binary_search.cpp:1]: This file already existed so, just continued with the previous one...
Thanks!

File: algorithms/C Language/BinarySearch.c [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Test 2:
+Enter the no of array elements: 6
+
+Enter the array elements: 1 4 6 9 23 45
+
+Enter the item to be searched: 6
+
+Element is found at index: 2
+
+*/
```
Comment (sobhanbera) [inline algorithms/C Language/BinarySearch.c:73]: Every outputs and input comment LGTM!

Thanks for your efforts!

**Review by sobhanbera — APPROVED** [review]
LGTM!

**Pull request merged**